core/src/data/rasterSource.cpp

```diff
@@ -2,58 +2,78 @@
 #include "data/propertyItem.h"
 #include "data/tileData.h"
 #include "tile/tile.h"
+#include "tile/tileBuilder.h"
 #include "tile/tileTask.h"
 #include "util/mapProjection.h"
-#include "platform.h"
+#include "log.h"
 
 namespace Tangram {
 
 class RasterTileTask : public BinaryTileTask {
 public:
-    RasterTileTask(TileID& _tileId, std::shared_ptr<TileSource> _source, int _subTask)
-        : BinaryTileTask(_tileId, _source, _subTask) {}
 
-    std::shared_ptr<Texture> m_texture;
+    const bool subTask = false;
+
+    std::unique_ptr<Texture> texture;
+    std::unique_ptr<Raster> raster;
```

Comment:
Looks like we can simplify this is only used to determine if the texture associated is "empty". Can we simplify this logic by just having a bool determining if the texture associated with the raster task is "empty"?

Reply:
Also does it make sense to add these "empty" textures to tile's rasters?

- Though to answer this question, might open a can of worms. I would suggest to document it and leave it for a subsequent PR.